lib/system/fatal.nim

```diff
@@ -42,7 +48,7 @@ elif (defined(nimQuirky) or defined(nimPanics)) and not defined(nimscript):
       add(buf, " [")
       add(buf, name exceptn)
       add(buf, "]\n")
-      cstderr.rawWrite buf
+      writeFatalErr(buf.cstring, buf.len)
```

Comment:
```suggestion
      writeToStdErr buf
```
I don't think there's a need to introduce `writeFatalErr`, instead `writeToStdErr` should be made to work cross platform

Reply:
I would prefer to reuse `writeToStdErr` but I couldn't find an elegant way to do it, `writeToStdErr` is mostly used in files that are `include`ed into `system.nim`, and as I recall `system/fatal` is imported or included in multiple locations.

Reply:
i think it's fixable, once you fix the other points i can look into how to fix that